src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -524,13 +529,21 @@ export function generateGenericAssayProfileDownloadData(
         // fall back to stableId if "NAME" not available
         downloadData.push(
             ['STUDY_ID', 'SAMPLE_ID'].concat(
-                _.map(stableIds, id =>
-                    getGenericAssayMetaPropertyOrDefault(
+                _.map(stableIds, id => {
+                    const entityName = getGenericAssayMetaPropertyOrDefault(
                         stableIdToMetaMap[id],
                         COMMON_GENERIC_ASSAY_PROPERTY.NAME,
                         id
-                    )
-                )
+                    );
+                    return GENERIC_ASSAY_CONFIG.genericAssayConfigByType[
+                        genericAssayType
+                    ]?.downloadTabConfig?.formatDownloadHeaderUsingCompactLabel
```

Comment:
this logic seems strange to me.  if it has this ```formatDownloadHeaderUsingCompactLabel``` then you call ```formatGenericAssayCompactLabelByNameAndId``` but it doesn't seem to use the first property?

Reply:
Yes, `formatDownloadHeaderUsingCompactLabel` is a boolean value in Generic Assay configuration. When it's true, we use `formatGenericAssayCompactLabelByNameAndId` to get a compact label